crates/astria-sequencer/src/service/consensus.rs

```diff
@@ -140,11 +139,42 @@ impl Consensus {
         Ok(response::InitChain::default())
     }
 
+    fn prepare_proposal(
```

Comment:
Can you add comments what exactly this does? I see that it generates the tx commitment, but I don't understand at first glance why it's prepended.